model/src/main/kotlin/licenses/DefaultLicenseInfoProvider.kt

```diff
@@ -82,7 +83,19 @@ class DefaultLicenseInfoProvider(
     private fun createDetectedLicenseInfo(id: Identifier): DetectedLicenseInfo {
         val findings = mutableListOf<Findings>()
 
-        ortResult.getScanResultsForId(id).forEach { (provenance, _, summary, _) ->
+        ortResult.getScanResultsForId(id).forEach { scanResult ->
+            var summary = scanResult.summary
+            var provenance = scanResult.provenance
+
+            if (provenance is RepositoryProvenance) {
+                val vcsPath = ortResult.getPackage(id)?.metadata?.vcsProcessed?.path?.orEmpty()!!
+
+                if ("$vcsPath/".startsWith("${provenance.vcsInfo.path}/")) {
```

Comment:
Add `vcsPath != provenance.vcsInfo.path && ...` to avoid unnecessary filtering if the path did not change.

Reply:
Also, shouldn't this append "/" only if the paths do not already end with a "/"?